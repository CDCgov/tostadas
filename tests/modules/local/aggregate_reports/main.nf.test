nextflow_process {

    name "Test Process AGGREGATE_REPORTS"
    script "modules/local/aggregate_reports/main.nf"
    process "AGGREGATE_REPORTS"

    test("Should aggregate one or more batch CSVs into a single report") {

        when {
            params {

            }

            process {
                """
                input[0] = [ 
                    file("${projectDir}/tests/modules/local/aggregate_reports/batch_1.csv")
                ]
                """
            }
        }

        then {
            assert process.success
            assert file("${process.out.submission_report[0]}").exists()
            assert snapshot(file("${process.out.submission_report[0]}")).match()
        }
    }
}
