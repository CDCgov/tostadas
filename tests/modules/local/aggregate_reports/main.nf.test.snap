{
    "Should aggregate one or more batch CSVs into a single report": {
        "content": [
            "submission_report.csv:md5,742a680a1878c92c807e6e00d5616b74"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:21:19.46947745"
    }
}